mesonbuild/compilers/detect.py

```diff
@@ -1042,6 +1042,10 @@ def detect_rust_compiler(env: 'Environment', for_machine: MachineChoice) -> Rust
             version = search_version(out)
 
             cls = rust.ClippyRustCompiler
+            mlog.warning(
```

Comment:
As the author of the clippy-driver support, please switch to using `FeatureDeprecated` for this so we can remove it in the future.

Reply:
This is a user feature not a developer one, so does it make sense to key it on `meson_version`? Anyhow I will look for a similar case and sort it out. :)

Reply:
Indeed, it doesn't make sense to key it on meson_version because it cannot be set from project files. At best, you could set it from a native file.

I am opposed to using FeatureDeprecated for this.

Reply:
I thinks it's appropriate since clippy-driver is discouraged for use by upstream and we now provide a better option that is supported by upstream.

@eli-schwartz, @jpakkane, @xclaesse: opinions on this?

Reply:
@dcbaker the point is that it's already a `WARNING: ...`, due to using `mlog.warning()`. The only difference that using FeatureDeprecated provides is making the `WARNING: ...` conditional on which minimum version the meson.build file requires due to the functions/syntax used in meson.build.

This is totally unrelated to "can we remove it in the future", which has never been the point of FeatureDeprecated.

Just change:
> `mlog.warning('clippy-driver is not intended as a general purpose compiler.  You can use "ninja clippy" in order to run clippy on a meson project.')`

to


> `mlog.warning('clippy-driver is not intended as a general purpose compiler.  You can use "ninja clippy" in order to run clippy on a meson project. Usee as a general purpose compiler will be removed in a future version of meson')`

and also frankly I don't see this as being covered by meson's stability guarantees. Projects which were written once, continue to work after upgrading meson.